src/modules/colorPicker/ColorPickerUI/Controls/ColorPickerControl.xaml.cs

```diff
@@ -288,22 +288,40 @@ private void ValueGradientSlider_ValueChanged(object sender, RoutedPropertyChang
 
         private void HexCode_TextChanged(object sender, TextChangedEventArgs e)
         {
+            bool insertHexTextHashtag = false;
             var newValue = (sender as TextBox).Text;
 
-            // support hex with 3 and 6 characters
-            var reg = new Regex("^#([0-9A-Fa-f]{3}){1,2}$");
+            // support hex with 3 or 6 characters and with or without #
+            var reg = new Regex("^#?([0-9A-Fa-f]{3}){1,2}$");
 
             if (!reg.IsMatch(newValue))
             {
                 return;
             }
 
+            // Add # if not typed by user. Prevents us form crash in converter.ConvertFromString()
+            if (!newValue.StartsWith("#", StringComparison.CurrentCulture))
+            {
+                newValue = "#" + newValue;
+                insertHexTextHashtag = true;
+            }
+
             if (!_ignoreHexChanges)
             {
+                _ignoreHexChanges = true;
+
+                // Update hex text box to show code with # and fix cursor position
+                if (insertHexTextHashtag)
+                {
+                    HexCode.Text = newValue;
+                    HexCode.Select(HexCode.Text.Length, 0);
+                    insertHexTextHashtag = false;
+                }
```

Comment:
@niels9001, @stefansjfw 
If you think we get problems with adding hashtag in real time while typing on slow systems we can think about adding it only when the user switches focus to another control.

Reply:
yes, I think the better way to do it either on focus to another control , or on flyout closing

Reply:
Will do it on focus change in the new pr.

Reply:
@stefansjfw 
Do you mean we get in trouble whe the user switches focus to the slider and changes it immediately? (By clicking near the dot and moving it immediately.)

Reply:
I don't think that will be a problem. But that's why I suggested to add # on flyout closing - it should be safer. We don't need to think about where would focus go and if it's going to affect the text field